substrate/client/rpc-spec-v2/src/chain_head/subscription/inner.rs

```diff
@@ -599,11 +591,6 @@ impl<Block: BlockT, BE: Backend<Block>> SubscriptionsInner<Block, BE> {
 		sub_id: String,
 		with_runtime: bool,
 	) -> Option<InsertedSubscriptionData<Block>> {
-		if self.suspend.is_suspended() {
-			log::trace!(target: LOG_TARGET, "[id={:?}] Subscription already suspended", sub_id);
-			return None
-		}
```

Comment:
Just to confirm; is the point of removing this now that we will allow new subscriptions to be added any time, but stop them all regardless whenever `generate_events` says that the distance is too large? Whereas before we'd not allow new subscriptions to be added for the timeout period.

Reply:
Yep, that makes sense

Reply:
Thanks! The changes def simplify things a fair bit so I'm in favour, and the block distance being too large is def an edge case anyways :)